File: java/src/test/java/com/penapereira/patterns/adapter/AdapterTest.java
package com.penapereira.patterns.adapter;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class AdapterTest {

    @Test
    void translatesRequestIntoSpecificRequest() {
        Target target = new Adapter(new Adaptee());
        assertEquals("Adapter(Adaptee)", target.request());
    }

    @Test
    void delegatesToWhicheverAdapteeItHolds() {
        Adaptee other = new Adaptee() {
            @Override
            public String specificRequest() {
                return "Other";
            }
        };
        assertEquals("Adapter(Other)", new Adapter(other).request());
    }
}

File: typescript/src/adapter/adapter.test.ts
import { describe, expect, it } from "vitest";
import { Adaptee } from "./adaptee.ts";
import { Adapter } from "./adapter.ts";
import type { Target } from "./target.ts";

describe("Adapter", () => {
  it("translates request() into specificRequest()", () => {
    const target: Target = new Adapter(new Adaptee());
    expect(target.request()).toBe("Adapter(Adaptee)");
  });

  it("delegates to whichever adaptee it holds", () => {
    class Other extends Adaptee {
      override specificRequest(): string {
        return "Other";
      }
    }
    expect(new Adapter(new Other()).request()).toBe("Adapter(Other)");
  });
});
